numba/cuda/compiler.py

```diff
@@ -74,6 +71,50 @@ def compile_kernel(pyfunc, args, link, debug=False, inline=False,
     return cukern
 
 
+@global_compiler_lock
+def compile_ptx(pyfunc, args, debug=False, fastmath=False, cc=None, opt=3):
+    """Compile a Python function to PTX for the given arguments.
+
+    :param pyfunc: The Python function to compile.
+    :param args: A tuple of argument types to compile for.
+    :param debug: Whether to include debug info in the generated PTX.
+    :type debug: bool
+    :param fastmath: Whether to enable fast math flags (ftz=1, prec_sqrt=0,
+                     prec_div=, and fma=1)
+    :type fastmath: bool
+    :param cc: Compute capability to compile for, as a tuple ``(MAJOR, MINOR)``.
+               Defaults to ``(5, 0)``.
+    :type cc: tuple
+    :param opt: Enable (``3``) or disable (``0``) optimizations. Defaults to
+                ``3``.
+    :type opt: int
+    :return: PTX code generated by NVVM.
```

Comment:
Perhaps an obvious question 😅: what's the encoding of the returned PTX code?

Reply:
I've generally been decoding it as `utf-8`, but I don't know for certain - will see if I can work out the definitive answer.

Reply:
I couldn't find an answer, but I've put decoding it inside the `compile_ptx` function, so it doesn't need to concern the user, and if that ever turns out not to work, we can treat it as a bug and fix it appropriately.